nomad/fsm.go

```diff
@@ -2450,6 +2452,36 @@ func (n *nomadFSM) applyHostVolumeDelete(msgType structs.MessageType, buf []byte
 	return nil
 }
 
+func (n *nomadFSM) applyTaskVolumeClaimRegister(msgType structs.MessageType, buf []byte, index uint64) interface{} {
```

Comment:
I suspect we won't want to register a claim outside of the allocation update, so we can get away without this FSM function. I think you're right that we'll want a delete in the FSM so that we can break associations via the API though.